plugins/src/android/OptimoveSDKPlugin.java

```diff
@@ -138,4 +275,155 @@ private void getCurrentUserIdentifier(CallbackContext callbackContext) {
             callbackContext.error("current user identifier is null");
         }
     }
+
+    private void setHandlersCallBackContext(CallbackContext callbackContext) {
+        jsCallbackContext = callbackContext;
+        PluginResult result = new PluginResult(PluginResult.Status.OK);
+        result.setKeepCallback(true);
+        callbackContext.sendPluginResult(result);
+    }
+
+    private void checkIfPendingPushExists() {
+        if (null != pendingPush) {
+            OptimoveSDKPlugin.sendMessageToJs("pushOpened",
```

Comment:
to trigger pushOpenedHandler, here we need jsCallbackContext set. Is there a possibility of race/sequencing issues? 

Now we add 2 listeners for 'deviceready' event: 
1) one in Optimove.js calls `init`, which sets `jsCallbackContext`
2) and one in index.js sets `pushOpenedHandler` which checks for pendingPush. 

Is there a guarantee that 1) runs before 2) ?

Reply:
updated to make it more robust racing-wise